x/programs/rust/examples/verify/src/types.rs

```diff
@@ -0,0 +1,84 @@
+use ed25519_dalek::VerifyingKey;
+use wasmlanche_sdk::types::Bytes32;
+
+// We create these custom wrapper types to allow us to pass them
+// across the WASM boundary.
+
+// EDVerifyingKey is a wrapper around ed25519_dalek::VerifyingKey.
+// this represents an ED25519 public key
+pub struct EDVerifyingKey(ed25519_dalek::VerifyingKey);
+
+// EDSignatureBytes represents the bytes of an ed25519 signature.
+pub struct EDSignatureBytes([u8; EDSignatureBytes::LEN]);
+
+// BatchPubKeys represents a batch of EDVerifyingKeys.
+// Currently, we need to hardcode the length of the batch(due to limitations of the SDK).
+// Once we have dynamic args, we can marshal the length in go and unmarshal in the macro.
+pub struct BatchPubKeys(pub Vec<EDVerifyingKey>);
+
+impl EDVerifyingKey {
+    const LEN: usize = ed25519_dalek::PUBLIC_KEY_LENGTH;
+    pub fn as_key(&self) -> &ed25519_dalek::VerifyingKey {
+        &self.0
+    }
+}
+
+impl EDSignatureBytes {
+    const LEN: usize = ed25519_dalek::SIGNATURE_LENGTH;
+    pub fn as_bytes(&self) -> &[u8; Self::LEN] {
+        &self.0
+    }
+}
+
+impl BatchPubKeys {
+    // hardcoded length of batch
+    const TEMP_LEN: usize = 5;
+    pub fn push(&mut self, pub_key: EDVerifyingKey) {
+        self.0.push(pub_key);
+    }
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+}
```

Comment:
Same comment about `Deref`. You want to get all the `slice` methods [the same way that `Vec` does](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#deref-methods-%5BT%5D)